firmware/application/apps/ui_settings.hpp

```diff
@@ -667,6 +672,47 @@ class SetConfigModeView : public View {
     };
 };
 
+using portapack::persistent_memory::fake_brightness_level_options;
+
+class SetFakeBrightnessView : public View {
+   public:
+    SetFakeBrightnessView(NavigationView& nav);
+
+    void focus() override;
+
+    std::string title() const override { return "Brightness"; };
+
+   private:
+    Labels labels{
+        {{1 * 8, 1 * 16}, "Cover a dark layer on screen;", Color::light_grey()},
+        {{1 * 8, 2 * 16}, "Influence a little performace", Color::light_grey()},
+        {{1 * 8, 3 * 16}, "if you enabled it.", Color::light_grey()},
+        {{2 * 8, 8 * 16}, "Brightness:", Color::light_grey()},
+    };
+
+    OptionsField field_fake_brightness{
+        {20 * 8, 8 * 16},
+        6,
+        {{"12.5%", fake_brightness_level_options::BRIGHTNESS_12p5},
+         {"25%", fake_brightness_level_options::BRIGHTNESS_25},
+         {"50%", fake_brightness_level_options::BRIGHTNESS_50},
+         {"100%", fake_brightness_level_options::BRIGHTNESS_100}}};
```

Comment:
Why do we need the 100% choice here?  Seems like that's the same thing as Off, except performance is slightly less.  I would just eliminate this one option from this OptionsField.  The rest of the code should be fine with that.

Reply:
Thanks! I fixed other request, and about this one, I have the following choices:
1. make the selection and enum and all other place all have 12.5, 25, 50. in this case, i'll add back the previous init line since it's not 0 anymore and i use the enum number at the shift opraation and i think't it's not worth to do a calculation again (+1).
2. just remove 100/0 in option field, keep the 0/100 in enum, but user would see weird behavior: didn't change anything and just click save but screen got dim.
3. keep it as it is now, but as you mentioned, we'll have a useless option.

which one is the best way? how do you think? thx!

Reply:
My recommendation is #1 (leaving the enum values equal to the bit shift same as you have now, 1=50%).

But, still I don't think the previously mentioned pmem init line is actually needed since the Settings code will handle the firmware upgrade case anyway where the value in pmem is currently 0 on every PortaPack.  To clarify, the following line in SetFakeBrightnessView will constrict the new brightness setting to one of the available valid brightness levels:

`field_fake_brightness.set_by_value(pmem::fake_brightness_level());`

(You might need to reorder the options though so that 50% is the value that it picks by default when it finds a 0 in pmem.)

Reply:
Note there is also a 4th option which would be to show the 100% option, but to implement it the same way as you're handing the "disabled" case.  If you did that, you could eliminate the enable/disable checkbox in SetFakeBrightnessView.  Then you would do something like this to configure the enable/disable flag in pmem:
`pmem::set_apply_fake_brightness(field_fake_brightness.selected_index_value() != BRIGHNESS_100);`

(This would require a bit more work though, including rewording of the text and perhaps modifying the title bar icon to perhaps cycle through the brightness levels versus on/off.)

Any votes for option 4?

Reply:
Option4 sounds like not that good (if I read it correctly) because I tend to keep way to save states (50% or 25% or so) and users wouldn’t lost the status after disabling it

Reply:
you are right! i fixed it as the way1 menthined.

Reply:
> Option4 sounds like not that good (if I read it correctly) because I tend to keep way to save states (50% or 25% or so) and users wouldn’t lost the status after disabling it

In option 4 that's true that firmware wouldn't remember the previous brightness level after it's turned off, but I don't know how much that might matter to the user.  I kinda like the idea of being able to click on the status bar to cycle through the 4 options (versus only between Off and some saved setting) without having to open an app.